### DISCUSSION TITLE
Strange behaviour and possible bug since starting to use Volar

### DISCUSSION THREAD

**Discussion by julie777** [vuejs/language-tools#560] [Q&A] [answered]
I recently switched to Volar in order to use <script setup>.
I am using eslint and prettier.

Periodically some things don't work and I haven't been able to determine if Volar is the culprit. Given that it started happening when I shifted to Volar it does seem likely. I have been turning off more and more, uninstalling all the other extensions, reducing the configuration in my vscode settings. Still no luck. 

If a file is messed up I can go to it and undo until it is back to normal but then when I format/save it works fine.
It is always in vue files.
It frequently happens when I save all. I'm not sure if it happens when I save an individual file.
My current settings have no actions on save, but format on save set. I'm not sure how all the vscode setting interact and how they interact with volar.

vscode settings.js - these are the user settings and I have created an empty file in the .vscode directory to avoid confusion.

```
 "editor.formatOnSave": true,
  "editor.copyWithSyntaxHighlighting": false,
  "editor.codeActionsOnSave": {},
  "editor.defaultFormatter": "esbenp.prettier-vscode",
  "[javascript]": {
    "editor.defaultFormatter": "esbenp.prettier-vscode"
  },
    "eslint.validate": [
    "json",
    "javascript",
    "javascriptreact",
    "typescript",
    "vue"
  ],
  "[vue]": {
    "editor.defaultFormatter": "johnsoncodehk.volar"
  },
  "volar.codeLens.pugTools": false,
  "prettier.semi": false,
  "prettier.singleQuote": true,
  "prettier.useEditorConfig": false,
  "typescript.validate.enable": false,
```

Problem details.
1. sometimes unused imports are removed, but not always
2. sometimes unreachable code is removed, but not always
3. sometimes random spaces are inserted or letters removed in both the template section and the script section.
![image](https://user-images.githubusercontent.com/5090005/136106554-b35c400e-a279-4f81-8d6d-f64e9cb6c9c4.png)
4. sometimes the compile output shows the results of 2. but the code doesn't have the spaces inserted.
![image](https://user-images.githubusercontent.com/5090005/136106367-f827432a-3290-476d-b663-546bb61917d4.png)

I would appreciate any help with this.

**Comment by johnsoncodehk** [comment, marked as answer]
> 1. sometimes unused imports are removed, but not always
> 2. sometimes unreachable code is removed, but not always

It should ESLint or other extension behavior, because you don't have code action on save, you can try disable ESLint to confirm.

> 3. sometimes random spaces are inserted or letters removed in both the template section and the script section.

I have note this problem and I think this is VSCode problem (https://github.com/vuejs/rfcs/pull/228#issuecomment-882207918) but not confirm. If you can provide a minimum reproduce I will survey it.

> 4. sometimes the compile output shows the results of 2. but the code doesn't have the spaces inserted.

Seems to be related to 3.? Not idea for now. :S
